### DISCUSSION TITLE
Code contributions should be highlighted in profiles, in contrast with creating issues or documentation-only pull requests

### DISCUSSION THREAD

**Discussion by marcospgp** [community/community#45469] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Contributions in Github profiles should highlight actual code changes in contrast with creating issues or documentation-only pull requests.

Contributing with code tends to require a lot more effort, and any incentive helps!

**Comment by ozcanyldzhn** [comment]
Highlighting actual code contributions separately from issue creation or documentation-only pull requests is an interesting idea. Here are a few potential improvements that could address this concern:

Enhanced Contribution Badges:

GitHub could introduce distinct badges or visual indicators in user profiles to differentiate code contributions from non-code contributions (e.g., issues, discussions, documentation updates).
Filterable Contribution Graph:

The contribution activity graph could include filters that allow users to view only code-related contributions separately from other types.
Weighted Contribution Score:

A scoring system could be introduced where code contributions carry more weight than other activities, giving a clearer representation of a developer's coding impact.
Improved Insights in Profile Activity:

The "Contributions" section could have a tab specifically for code-related pull requests to distinguish them more clearly.
This would provide better visibility into actual coding efforts while still valuing other forms of contributions.

Would love to hear more thoughts on how this could be implemented!
